.cursor/rules/005-clean-code.mdc

```
@@ -1,70 +1,80 @@
+### Core Principles
+- [YAGNI](mdc:.cursor/rules/0057-clean-code-yagni.mdc)
+- [DRY](mdc:.cursor/rules/0058-clean-code-dry.mdc)
+- [SOLID](mdc:.cursor/rules/0059-clean-code-solid.mdc)
+
+### Testing and Quality
+- [Testing Standards](mdc:.cursor/rules/00510-clean-code-tests.mdc)
+- [Code Smells](mdc:.cursor/rules/00511-clean-code-smells.mdc)
 
 ### Code Organization
-- One file per feature, split responsibility across files (SRP).
-- Keep files small (under 300 lines)
-- When folders exceed 7 files, create sub-folders
-- Group related code together
-- Follow the principle of least surprise
-- Maintain a consistent level of abstraction within functions
-- Group code by business domain rather than technical layer
-- Focus on functional specifications over technical specifications
-- Keep business logic separate from technical implementation
+- [Understanding](mdc:.cursor/rules/0051-clean-code-understanding.mdc)
+- [Naming](mdc:.cursor/rules/0052-clean-code-naming.mdc)
+- [Functions](mdc:.cursor/rules/0053-clean-code-functions.mdc)
+- [Comments](mdc:.cursor/rules/0054-clean-code-comments.mdc)
+- [Structure](mdc:.cursor/rules/0055-clean-code-structure.mdc)
+- [Objects](mdc:.cursor/rules/0056-clean-code-objects.mdc)
 
-### Comments
-- Write self-documenting code that minimizes the need for comments
-- Use comments to explain "why" not "what" or "how"
-- Keep comments up-to-date with code changes
+## Examples
+<example>
+// Good: Following core principles
+class EmailService {
+  constructor(private readonly emailProvider: IEmailProvider) {}
+  
+  async sendEmail(email: Email): Promise<void> {
+    await this.emailProvider.send(email);
+  }
+}
+</example>
 
-### Error Handling
-- Use custom domain error for different error categories
-- Error messages must be meaningful and SHOULD be translated to the user's language (e.g. French) for display.
-- Fail fast: early throws errors, avoid returning null or undefined
```

Fail fast: throw errors early